gpii/node_modules/gpii-oauth2/gpii-oauth2-authz-server/public/src/components/privacySettings/js/EditPrivacySettings.js

```diff
@@ -0,0 +1,241 @@
+            },
+            "afterRender.bindDone": {
+                "this": "{that}.dom.done",
+                method: "click",
+                args: "{that}.savePrefsAndExit"
+            }
+        },
+        invokers: {
+            closeDialog: {
+                "this": "{that}.dialog",
+                method: "dialog",
+                args: ["close"]
+            },
+            fetchDecisionPrefs: {
+                funcName: "gpii.oauth2.ajax",
+                args: ["{that}.options.requestInfos.fetchDecisionPrefs.url", {
+                    authDecisionId: "{that}.model.clientData.authDecisionId"
+                }, {
+                    type: "{that}.options.requestInfos.fetchDecisionPrefs.type",
+                    dataType: "json",
+                    success: "{that}.setDecisionPrefs"
+                }]
+            },
+            saveDecisionPrefs: {
+                funcName: "gpii.oauth2.ajax",
+                args: ["{that}.options.requestInfos.saveDecisionPrefs.url", {
+                    authDecisionId: "{that}.model.clientData.authDecisionId"
+                }, {
+                    type: "{that}.options.requestInfos.saveDecisionPrefs.type",
+                    dataType: "json",
+                    contentType: "application/json",
+                    data: {
+                        expander: {
+                            funcName: "JSON.stringify",
+                            args: [{
+                                expander: {
+                                    func: "{selectionTree}.toServerModel",
+                                    args: "{selectionTree}.model"
+                                }
+                            }]
+                        }
+                    }
+                }]
+            },
+            fetchAvailableAuthorizedPrefs: {
+                funcName: "gpii.oauth2.ajax",
+                args: ["{that}.options.requestInfos.fetchAvailableAuthorizedPrefs.url", {
+                    clientID: "{that}.model.clientData.oauth2ClientId"
+                }, {
+                    //TODO: Handle errors
+                    dataType: "json",
+                    success: "{that}.setAvailableAuthorizedPrefs"
+                }]
+            },
+            setAvailableAuthorizedPrefs: {
+                changePath: "availableAuthorizedPrefs",
+                value: "{arguments}.0"
+            },
+            composeRequestURL: {
+                funcName: "fluid.stringTemplate",
+                args: ["{arguments}.0", {
+                    authDecisionId: "{that}.model.clientData.authDecisionId"
+                }]
+            },
+            setDecisionPrefs: {
+                changePath: "decisionPrefs",
+                value: "{arguments}.0"
+            },
+            savePrefsAndExit: {
+                funcName: "gpii.oauth2.editPrivacySettings.savePrefsAndExit",
+                args: ["{that}"]
+            },
+            setDoneState: {
```

Comment:
I've filed a JIRA issue for this work:

http://issues.gpii.net/browse/GPII-1124